applications: serial_lte_modem: Add SMS support (#4610)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/SMS_AT_commands.rst

.. _SLM_AT_SMS:

SMS AT commands
****************

.. contents::
   :local:
   :depth: 2

The following commands list contains SMS-related AT commands.

SMS support #XSMS
=================

> The ``#XSMS`` command support *SMS Send and Receive* functionalities.

Reviewer: ```suggestion
The ``#XSMS`` command supports functionalities for sending and receiving SMS messages.
```

Set command
-----------

The set command allows you to start or stop SMS, as well as send SMS text.
Only GSM-7 encoding is supported, 8-bit and UCS2 encodings are not supported.

Syntax
~~~~~~

::

   #XSMS=<op>[,<number>,<text>]

* The ``<op>`` parameter can accept one of the following values:

  * ``0`` - Stop SMS
  * ``1`` - Start SMS, ready to receive
  * ``2`` - Send SMS

* The ``<number>`` parameter is a string.

>   It represents the SMS recipient's phone number, including the country code (for example *+81xxxxxxx*).

Reviewer: ```suggestion
  It represents the SMS recipient's phone number, including the country code (for example ``+81xxxxxxx``).
```

* The ``<text>`` parameter is  a string.
  It is the SMS text to be sent.

> Unsolicited Notification

Reviewer: ```suggestion
Unsolicited notification
```

~~~~~~~~~~~~~~~~~~~~~~~~

This is the notification syntax when an SMS message is received:

  ::

      #XSMS: <datetime>,<number>,<text>

  * The ``<datetime>`` value is a string.

>     It represents when the SMS is received.

Reviewer: ```suggestion
    It represents the time when the SMS is received.
```

.. _SLM_AT_SMS:

SMS AT commands
****************

.. contents::
   :local:
   :depth: 2

The following commands list contains SMS related AT commands.

SMS support #XSMS
=================

The ``#XSMS`` command support SMS Send and Receive.

Set command
-----------

The set command allows you to start or stop SMS, as well as send SMS text.
Only GSM-7 encoding is supported, 8-bit and UCS2 encodings are not supported.

Syntax
~~~~~~

::

   #XSMS=<op>[,<number>,<text>]

* The ``<op>`` parameter can accept one of the following values:

  * ``0`` - Stop SMS
  * ``1`` - Start SMS, ready to receive
  * ``2`` - Send SMS

* The ``<number>`` parameter is a string.
  It represents SMS recipient's phone number.

>   NOTE some SMS service providers require this to be international type. E.g. "+81xxxxxxx"

Author: I'll add it as I'm going to patch SMS lib a bit as well

>   NOTE some SMS service providers require this to be international type. E.g. "+81xxxxxxx"

Reviewer 2: Ok
